go-runtime/compile/build.go

```diff
@@ -671,7 +700,41 @@ func getSumTypes(module *schema.Module, sch *schema.Schema, nativeNames NativeNa
 	slices.SortFunc(out, func(a, b goSumType) int {
 		return strings.Compare(a.Discriminator, b.Discriminator)
 	})
-	return out
+
+	var externalTypes []goExternalType
+	for im, types := range goExternalTypes {
+		externalTypes = append(externalTypes, goExternalType{
+			Import: im,
+			Types:  types,
+		})
+	}
+	return out, externalTypes
+}
+
+func getGoExternalType(fqName string) (_import string, _type string) {
+	parts := strings.Split(fqName, "/")
+	pkgParts := strings.Split(parts[len(parts)-1], ".")
+	// expect 2 parts when package and directory path are the same, e.g. "github.com/foo/bar.A"
+	// expect 3 parts when package and directory path differ, e.g. "github.com/foo/bar.baz.A"
+	if l := len(pkgParts); !(l == 2 || l == 3) {
+		panic("unexpected Go qualified name format: " + fqName)
+	}
+
+	// last part is the type name
+	name := pkgParts[len(pkgParts)-1]
+	// second to last part is the package name (in both cases)
+	pkg := pkgParts[len(pkgParts)-2]
+	// translate the fqName to a valid import; removing the type name (and package name if different from directory)
+	im := fmt.Sprintf("%q", strings.TrimSuffix(fqName, parts[len(parts)-1])+pkgParts[0])
+	if len(pkgParts) == 3 {
+		// this means pkg and directory path differ, e.g. "github.com/foo/bar.baz.A" where the directory is "bar"
+		// and the package is "baz".
+		//
+		// in this case, import has an alias with the real package name: `import baz "github.com/foo/bar"`
+		im = fmt.Sprintf("%s %s", pkg, im)
```

Comment:
This stuff is pretty hairy! ^

Reply:
i know 😬 lemme revisit and maybe clarify it with some more variable names